### ISSUE TITLE
Comment Template: Improve selection for the inner blocks

### ISSUE THREAD

**Issue by gziolo** [WordPress/gutenberg#37154]
### Description

Originally reported in https://github.com/WordPress/gutenberg/pull/36065#pullrequestreview-821372216:

> Selecting blocks for nested comments sometimes changes the position on the screen in an unexpected way, you can observe that in the video I included:
>
> https://user-images.githubusercontent.com/699132/144412452-0e71915c-eea4-4e67-b9e2-4e3ef4114c85.mov

Confirmed in https://github.com/WordPress/gutenberg/pull/36065#issuecomment-985146916 by @michalczaplinski:

> There is some issue with block focus/selection, related to what you mentioned in your last point. Sometimes, you have to click twice or three times in order to select the relevant block inside of the Template. And sometimes. even then, the an incorrect block gets selected initially.

It might be the same issue that @andrewserong is trying to resolve for Query Loop and Post Comment in https://github.com/WordPress/gutenberg/pull/36431.

### Step-by-step reproduction instructions

1. Insert the Comments Query Loop block in the site template and make sure that it contains comments with replies.
2. Try to select inner blocks in the Comment Template block for the nested (replies) comments.

### Screenshots, screen recording, code snippet

https://user-images.githubusercontent.com/699132/144412452-0e71915c-eea4-4e67-b9e2-4e3ef4114c85.mov

### Environment info

- `trunk` in Gutenberg
- WordPress 5.9 beta 1
- Safari on macOS

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by andrewserong** [comment]
> It might be the same issue that @andrewserong is trying to resolve for Query Loop and Post Comment in #36431.

Thanks for the ping! That PR seeks to address part of the problem (preserve the wide layout alignment, remove flicker / unintentional scroll when switching the current block context). There's another problem to deal with, that I thought could be good for a follow-up, which is to see if it's possible in the click handler when switching block contexts, to also then set the currently selected block to the one that the user clicked on within the block preview. I'm not quite sure how feasible it is to do that neatly, so thought it'd make a good potential follow-up if that PR winds up landing. (Happy to explore within that PR if folks think it's better to try to deal with it all at once, of course!)

**Comment by gziolo** [comment]
A quick update, I have just merged https://github.com/WordPress/gutenberg/pull/36431 that radically improves the experience around the block selection for the Query Loop block. 

> There's another problem to deal with, that I thought could be good for a follow-up, which is to see if it's possible in the click handler when switching block contexts, to also then set the currently selected block to the one that the user clicked on within the block preview.

It's a very tricky issue we deal with here because we mostly show block previews but at the same time we have to support full interactivity for the currently selected block. Let us know when you have a prototype ready so we can stay in the loop.

**Comment by andrewserong** [comment]
> It's a very tricky issue we deal with here because we mostly show block previews but at the same time we have to support full interactivity for the currently selected block. Let us know when you have a prototype ready so we can stay in the loop.

Will do! Thanks for approving and merging in #36431 for me — now that we have that PR in, it should be easier to focus on this problem in isolation 🤞

**Comment by gziolo** [comment]
The current approach we have has more drawbacks. Sometimes it renders the same controls multiple times:

<img width="1183" alt="Screenshot 2021-12-21 at 13 09 20" src="https://user-images.githubusercontent.com/699132/146927909-0bb436d6-b81b-4a04-a1dc-8ebe1fc7cd8a.png">

**Comment by andrewserong** [comment]
I'm not sure if this helps, but I ran into a similar issue in my WIP PR in #37519 for the Post Template block when I removed `pointer-events: none` on the block preview — this meant that when clicking between the Query loop previews, the clicking on a block within the preview appeared to render the component and its InspectorControls fill, for the particular clientId (given that the clientId of the preview is the same for each instance of the loop). I _think_ the `pointer-events: none` there (and earlier in the `Disabled` component) typically prevents this in the live preview, but I'm curious if there's a way to prevent the components within a block preview from ever rendering its slotfills / live behaviour.

Apologies if that doesn't _quite_ make sense — I'm a little fuzzy on how those pieces interact. I haven't noticed the above issue in the Query Loop (not comments) block in testing on `trunk`, though.
